api/hub/MODERATION_API.md

```diff
@@ -0,0 +1,565 @@
+  "creationDate": 1631018869481,
+  "delisted": true,
+  "explanation": "This had to be moderated.",
+  "moderated": true,
+  "moderatedDate": 1631262937970,
+  "moderator": "byareibr3tw...lnmuh3x5irpdbjy"
+}
+```
+
+**Response codes:**
+ * HTTP 200 (OK) for a successful request.
+ * HTTP 500 (Internal server error) for any other backend-related errors.
+
+
+## Add a moderator
+
+This feature requires admin priviledges (currently token-based). It assigns moderator status 
+(and optionally admin status) to the provided user.
+
+**secret** - a secret token to authenticate the request
+**data** - the request (`user`, `admin`, `active`)
+**user** - the user identifier
+**admin** - whether to make this user an admin (boolean)
+**active** - whether to make this user active immediately (boolean)
+
+```
+POST https://api.ethereum.world/api/moderation/moderators/new
+Content-Type: application/json
+{
+  "secret": "foo",
+  "data": { 
+    "user": "byareibr3tw...lnmuh3x5irpdbjy",
+    "admin": true,
+    "active": true 
+  }
+}
+```
+
+**Response codes:**
+ * HTTP 200 (OK) for a successful request.
+ * HTTP 400 (Bad request) if missing `data` or `secret` from the request.
+ * HTTP 403 (Forbidden) if the user is not allowed (the secret does not match).
+ * HTTP 500 (Internal server error) for any other backend-related errors.
+
+## Update a moderator
+
+The following attributes can be updated:
+
+ * `admin` (true|false)- whether the user has admin status
+ * `active` (true|false) - whether the account is active or not
+
+
+```
+POST https://api.ethereum.world/api/moderation/moderators/byareibr3tw...lnmuh3x5irpdbjy
+Content-Type: application/json
+{
+  "secret": "foo",
+  "data": { 
+    "admin": true,
+    "active": false 
+  }
+}
+```
+
+**Response codes:**
+ * HTTP 200 (OK) for a successful request.
+ * HTTP 400 (Bad request) if missing `user`, `data` or `secret` from the request.
+ * HTTP 403 (Forbidden) if the user is not allowed (the secret does not match).
+ * HTTP 500 (Internal server error) for any other backend-related errors.
+
+
+## Check if the given ETH address is _currently_ a moderator
+
+Easily check if the user is a moderator.
+
+```
+HEAD https://api.ethereum.world/api/moderation/moderators/status/byareibr3tw...lnmuh3x5irpdbjy
+```
+
+**Response codes:**
+ * HTTP 200 (OK) for a successful request stating that the user is indeed a moderator.
+ * HTTP 404 (Not found) if the ETH address owner is not a moderator or is a disabled account.
+
+## Get data for a given moderator.
+
+```
+GET https://api.ethereum.world/api/moderation/moderators/status/byareibr3tw...lnmuh3x5irpdbjy
+
+{
+  "_id": "byareibr3tw...lnmuh3x5irpdbjy",
+  "_mod": 1628581547366119200,
+  "active": true,
+  "admin": true,
+  "creationDate": 1628581547020
+}
+```
+
+**Response codes:**
+ * HTTP 200 (OK) for a successful request stating that the user is indeed a moderator.
+ * HTTP 404 (Not found) if the ETH address owner is not a moderator or is a disabled account.
```

Comment:
For clarity, I think we can replace all mentions of `ETH address` with `PubKey` in the doc for app related moderation requests since we are are now using PubKey for queries.